tubular/mapping.py

```diff
@@ -35,40 +34,58 @@ class BaseMappingTransformer(BaseTransformer):
     """
 
     def __init__(self, mappings, **kwargs):
-
         if isinstance(mappings, dict):
-
             if not len(mappings) > 0:
-
                 raise ValueError(f"{self.classname()}: mappings has no values")
 
             for j in mappings.values():
-
                 if not isinstance(j, dict):
-
                     raise ValueError(
                         f"{self.classname()}: values in mappings dictionary should be dictionaries"
                     )
 
             self.mappings = mappings
 
         else:
-
             raise ValueError(f"{self.classname()}: mappings must be a dictionary")
 
         columns = list(mappings.keys())
 
         super().__init__(columns=columns, **kwargs)
 
-    def transform(self, X):
-        """Base mapping transformer transform method.  Checks that the mappings
+    def check_dtype_changes(self, X, original_dtypes, suppress_dtype_warning=False):
+        mapped_columns = self.mappings.keys()
+        mapped_dtypes = X[mapped_columns].dtypes
+
+        if not suppress_dtype_warning:
```

Comment:
It might be better to bring `if not suppress_dtype_warning` out of this function.  Since the check might slow the transformer down it would be good to both skip the logging of the mappings and the check itself if suppress_dtype_warning is True.  Also cleaner code to have less args passing.